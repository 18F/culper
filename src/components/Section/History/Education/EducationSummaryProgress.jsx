import React from 'react'
import PropTypes from 'prop-types'

import { i18n } from 'config'

import SummaryCounter from 'components/Section/History/SummaryCounter'
import { totalYears } from 'components/Section/History/helpers'

import { EducationItemValidator } from 'validators'

const schoolRangesList = (items) => {
  const dates = []

  items.forEach((i) => {
    if (!i.Item || !i.Item.Dates || !i.Item.Dates.to || !i.Item.Dates.from) {
      return
    }

    if (new EducationItemValidator(i.Item).isValid()) {
      dates.push(i.Item.Dates)
    }
  })

  return dates
}

const diplomasRangesList = (items) => {
  const dates = []

  items.forEach((i) => {
    if (!i.Item) { return }

    if (!new EducationItemValidator(i.Item).isValid()) {
      return
    }

    if (i.Item.Diplomas.items) {
      i.Item.Diplomas.items.forEach((d) => {
        if (!d.Item || !d.Item.Date) { return }

        dates.push(d.Item.Date)
      })
    }
  })

  return dates
}

const EducationSummaryProgress = (props) => {
  const { Education, Birthdate, years } = props

  let schoolDates = []
  if (Education && Education.List && Education.List.items) {
    schoolDates = Education.List.items
  }

  const getSchoolDates = () => schoolRangesList(schoolDates)
  const getDiplomaDates = () => diplomasRangesList(schoolDates)

  return (
    <SummaryCounter
      className="education"
      title={i18n.t('history.education.summary.title')}
      schools={getSchoolDates}
      diplomas={getDiplomaDates}
      schoolsLabel={i18n.t('history.education.summary.schools')}
      diplomasLabel={i18n.t('history.education.summary.diplomas')}
      total={totalYears(Birthdate, years)}
    />
  )
}

/* eslint react/forbid-prop-types: 0 */
EducationSummaryProgress.propTypes = {
  Education: PropTypes.object,
  Birthdate: PropTypes.any,
  years: PropTypes.number,
}

EducationSummaryProgress.defaultProps = {
  Education: null,
  Birthdate: null,
  years: 10,
}

export default EducationSummaryProgress
