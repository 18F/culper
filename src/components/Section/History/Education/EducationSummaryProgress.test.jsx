import React from 'react'
import { shallow } from 'enzyme'

import EducationSummaryProgress from './EducationSummaryProgress'

describe('The EducationSummaryProgress component', () => {
  it('renders without errors', () => {
    const component = shallow(<EducationSummaryProgress />)

    expect(component.exists()).toBe(true)
    expect(component).toMatchSnapshot()
  })
})
