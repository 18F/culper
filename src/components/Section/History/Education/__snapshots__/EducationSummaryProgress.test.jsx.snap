// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EducationSummaryProgress component renders without errors 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SummaryCounter
    className="education"
    diplomas={[Function]}
    diplomasLabel="Degrees/Diplomas earned"
    schools={[Function]}
    schoolsLabel="Schools attended"
    title="Where you went to school"
    total={10}
  >
    <div
      className="summary-icon"
    >
      <Svg
        alt="Years covered for schools attended"
        src="/img/school-cap.svg"
      />
    </div>
  </SummaryCounter>,
  "nodes": Array [
    <SummaryCounter
      className="education"
      diplomas={[Function]}
      diplomasLabel="Degrees/Diplomas earned"
      schools={[Function]}
      schoolsLabel="Schools attended"
      title="Where you went to school"
      total={10}
    >
      <div
        className="summary-icon"
      >
        <Svg
          alt="Years covered for schools attended"
          src="/img/school-cap.svg"
        />
      </div>
    </SummaryCounter>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <EducationSummaryProgress
        Birthdate={null}
        Education={null}
        years={10}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "Birthdate": null,
          "Education": null,
          "years": 10,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SummaryCounter
          className="education"
          diplomas={[Function]}
          diplomasLabel="Degrees/Diplomas earned"
          schools={[Function]}
          schoolsLabel="Schools attended"
          title="Where you went to school"
          total={10}
        >
          <div
            className="summary-icon"
          >
            <Svg
              alt="Years covered for schools attended"
              src="/img/school-cap.svg"
            />
          </div>
        </SummaryCounter>,
        "_debugID": 2,
        "_renderedOutput": <SummaryCounter
          className="education"
          diplomas={[Function]}
          diplomasLabel="Degrees/Diplomas earned"
          schools={[Function]}
          schoolsLabel="Schools attended"
          title="Where you went to school"
          total={10}
        >
          <div
            className="summary-icon"
          >
            <Svg
              alt="Years covered for schools attended"
              src="/img/school-cap.svg"
            />
          </div>
        </SummaryCounter>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <EducationSummaryProgress
    Birthdate={null}
    Education={null}
    years={10}
  />,
}
`;
